lib/cereal/builders/entity.ex

```diff
@@ -27,18 +27,19 @@ defmodule Cereal.Builders.Entity do
 
   # if an assigns/2 function was specified in the serializer - we'll add its
   # return value to the conn's assigns
-  defp do_assigns(%{serializer: serializer, conn: %Plug.Conn{} = conn} = context) do
+  defp do_assigns(%{serializer: serializer, conn: conn, data: data} = context) do
+    assigns = Map.get(conn, :assigns) || %{}
+    conn = Map.put(conn, :assigns, assigns)
```

Comment:
I think this might be a good case for put_new?

```elixir
conn = Map.put_new(conn, :assigns, %{})
```

Reply:
oh nice! 🙏